python/mxnet/ndarray/ndarray.py

```diff
@@ -286,6 +286,10 @@ def __eq__(self, other):
         """x.__eq__(y) <=> x==y <=> mx.nd.equal(x, y) """
         return equal(self, other)
 
+    def __hash__(self):
+        """Default hash function."""
+        return id(self)//16
```

Comment:
why //16?

Reply:
This happens to be the default hash function for python objects when `__eq__` is not defined.

Reply:
Does this still apply?


You should add a comment to this, it's unclear.

In [1]: import mxnet as mx
^[[A
In [2]: b=mx.nd.array([1,2,3,4])

In [3]: a=mx.nd.array([1,2,3,4,5])

In [4]: hex(id(a))
Out[4]: '0x11d39df60'

In [5]: hex(id(b))
Out[5]: '0x11d383c88'


https://stackoverflow.com/questions/11324271/what-is-the-default-hash-in-python

Reply:
@larroy why does it matter?

Reply:
Is the division by 16 needed? If yes i would comment on why. If not, I would just remove it. It causes confusion to use such a constant without further explanation or a clarifying comment.